pkg/bootstrapper/bootstrapper_test.go

```diff
@@ -107,3 +113,157 @@ func TestPrepKubeletConfForWindows(t *testing.T) {
+
+	// Create a temp directory with wmcb prefix
+	dir, err := ioutil.TempDir("", "wmcb")
+	require.NoError(t, err, "error creating temp directory")
+	// Ignore the return error as there is not much we can do if the temporary directory is not deleted
+	defer os.RemoveAll(dir)
+
+	wnb := winNodeBootstrapper{
+		installDir:  dir,
+		kubeletArgs: make(map[string]string),
+	}
+
+	err = wnb.parseIgnitionFileContents([]byte(ignitionContents), map[string]fileTranslation{})
+	assert.NoError(t, err, "error parsing ignition file contents")
+	assert.FileExists(t, path.Join(dir, "cloud.conf"), "cloud.conf was not created")
+
+	confContents, err := ioutil.ReadFile(path.Join(dir, "cloud.conf"))
+	assert.NoError(t, err, "error reading cloud.conf")
+
+	conf := string(confContents)
+	// Check if the file beings with { and ends with }
+	assert.True(t, strings.HasPrefix(conf, "{"))
+	assert.True(t, strings.HasSuffix(conf, "}"))
+
+	// Replace the beginning {\n\t, \n}, with ""
+	conf = strings.Replace(conf, "{\n\t", "", -1)
+	conf = strings.Replace(conf, "\n}", "", -1)
+
+	// Split the conf items into an array. Each element will now contain "key: value"
+	confItems := strings.Split(conf, ",\n\t")
+
+	// Expected key value pairs from ignitionContents
+	confExpected := map[string]string{
+		"cloud":             "AzurePublicCloud",
+		"tenantId":          "1234a1b2-a1bc-123a-123a-ab1c2de3afgh",
+		"subscriptionId":    "1a123456-12ab-123a-1234-abc1d1ab01c0",
+		"resourceGroup":     "winc-test-rg",
+		"location":          "centralus",
+		"vnetName":          "winc-test-vnet",
+		"vnetResourceGroup": "winc-test-rg",
+		"subnetName":        "winc-test-node-subnet",
+		"securityGroupName": "winc-test-node-nsg",
+		"routeTableName":    "winc-test-node-routetable",
+	}
+
+	for _, confItem := range confItems {
+		// keyValue will have two elements, 0 being the key and 1 the value
+		keyValue := strings.Split(confItem, ":")
+		assert.True(t, len(keyValue) == 2)
+
+		// Check if the key needs to be compared
+		value, present := confExpected[keyValue[0]]
+		if !present {
+			continue
+		}
+
+		// Assert that the key value from the file matches the value in the ignition contents
+		assert.Equal(t, confExpected[keyValue[0]], value)
+	}
+
+	// Check that the --cloud-conf option value is present in the kubelet args and matches tempdir + /cloud.conf
+	cloudConfigOptValue, present := wnb.kubeletArgs["cloud-config"]
+	assert.True(t, present, "cloud-config option is not present in kubelet args")
+	assert.Equal(t, path.Join(dir, "cloud.conf"), cloudConfigOptValue,
+		"unexpected --cloud-config value %s", cloudConfigOptValue)
+}
+
+// TestCloudConfNotPresent tests if parseIgnitiionFileContents does not create a cloud.conf file if it is not present in
```

Comment:
s/parseIgnitiionFileContents/parseIgnitionFileContents

Reply:
All the negatives in this comment made it hard for me to figure out what was being tested, and I had to read the test to understand. To me, something like this is easier to read:
`TestCloudConfNotPresent tests that parseIgnitionFileContents will only create a cloud.conf file and add the "--cloud-config" option to the kubelet args, if the cloud.conf file is present in the ignition file.`